tests/impl.cpp

```diff
@@ -98,835 +190,56 @@ static inline float bankersRounding(float val)
-    ASSERT_RETURN(t[8] == u8);
-    ASSERT_RETURN(t[9] == u9);
-    ASSERT_RETURN(t[10] == u10);
-    ASSERT_RETURN(t[11] == u11);
-    ASSERT_RETURN(t[12] == u12);
-    ASSERT_RETURN(t[13] == u13);
-    ASSERT_RETURN(t[14] == u14);
-    ASSERT_RETURN(t[15] == u15);
-    return true;
-}
-
-bool validateSingleFloatPair(float a, float b)
-{
-    const uint32_t *ua = (const uint32_t *) &a;
-    const uint32_t *ub = (const uint32_t *) &b;
-    // We do an integer (binary) compare rather than a
-    // floating point compare to take nands and infinities
-    // into account as well.
-    return (*ua) == (*ub);
-}
-
-bool validateSingleDoublePair(double a, double b)
-{
-    const uint64_t *ua = (const uint64_t *) &a;
-    const uint64_t *ub = (const uint64_t *) &b;
-    // We do an integer (binary) compare rather than a
-    // floating point compare to take nands and infinities
-    // into account as well.
-    return (*ua) == (*ub);
-}
-
-bool validateFloat(__m128 a, float f0, float f1, float f2, float f3)
-{
-    const float *t = (const float *) &a;
-    ASSERT_RETURN(validateSingleFloatPair(t[0], f0));
-    ASSERT_RETURN(validateSingleFloatPair(t[1], f1));
-    ASSERT_RETURN(validateSingleFloatPair(t[2], f2));
-    ASSERT_RETURN(validateSingleFloatPair(t[3], f3));
-    return true;
-}
-
-bool validateFloatEpsilon(__m128 a,
-                          float f0,
-                          float f1,
-                          float f2,
-                          float f3,
-                          float epsilon)
-{
-    const float *t = (const float *) &a;
-    float df0 = fabsf(t[0] - f0);
-    float df1 = fabsf(t[1] - f1);
-    float df2 = fabsf(t[2] - f2);
-    float df3 = fabsf(t[3] - f3);
-    ASSERT_RETURN(df0 < epsilon);
-    ASSERT_RETURN(df1 < epsilon);
-    ASSERT_RETURN(df2 < epsilon);
-    ASSERT_RETURN(df3 < epsilon);
-    return true;
-}
-
-bool validateDouble(__m128d a, double d0, double d1)
-{
-    const double *t = (const double *) &a;
-    ASSERT_RETURN(validateSingleDoublePair(t[0], d0));
-    ASSERT_RETURN(validateSingleDoublePair(t[1], d1));
-    return true;
-}
-
-bool test_mm_setzero_si128(void)
+result_t test_mm_setzero_si128(const SSE2NEONTestImpl &impl, uint32_t i)
 {
     __m128i a = _mm_setzero_si128();
     return validateInt32(a, 0, 0, 0, 0);
 }
 
-bool test_mm_setzero_ps(void)
+result_t test_mm_setzero_ps(const SSE2NEONTestImpl &impl, uint32_t i)
 {
     __m128 a = _mm_setzero_ps();
     return validateFloat(a, 0, 0, 0, 0);
 }
 
-bool test_mm_set1_ps(float w)
+result_t test_mm_set1_ps(const SSE2NEONTestImpl &impl, uint32_t i)
 {
+    float w = impl.mTestFloats[i];
     __m128 a = _mm_set1_ps(w);
     return validateFloat(a, w, w, w, w);
 }
 
-bool test_mm_set_ps(float x, float y, float z, float w)
+result_t test_mm_set_ps(const SSE2NEONTestImpl &impl, uint32_t i)
 {
+    float x = impl.mTestFloats[0];
+    float y = impl.mTestFloats[1];
+    float z = impl.mTestFloats[2];
+    float w = impl.mTestFloats[3];
```

Comment:
I think this should be replaced with `impl.mTestFloats[i]` or `impl.mTestFloatPointer1[0]`.
Otherwise, it would keep using the same value.